scopes/component/isolator/capsule-list.ts

```diff
@@ -27,6 +29,38 @@ export default class CapsuleList extends Array<Capsule> {
   getAllComponents(): Component[] {
     return this.map((c) => c.component);
   }
+  // Sort the capsules by their dependencies. The capsules with no dependencies will be first. Returns a new array.
+  async toposort(host: ComponentFactory): Promise<CapsuleList> {
+    const components = this.getAllComponents();
+    const legacyIdStrToComponentId: { [bitIdStr: string]: ComponentID } = {};
+    const graph = new Graph<Component, string>();
```

Comment:
building the graph shouldn't be a capsule list responsibility.
it's part of the component factory interface (so scope/workspace)
even better - use the graph aspect to handle it.
you can if you want to add more options to the graph creation in case the graph isn't exactly what you need.
you can maybe add another API to build a graph from components, not only from ids if you want.